dashboard/test/models/lti_user_identity_test.rb

```diff
@@ -6,4 +6,10 @@ class LtiUserIdentityTest < ActiveSupport::TestCase
     refute build(:lti_user_identity, user: nil).valid? "user is required"
     refute build(:lti_user_identity, lti_integration: nil).valid? "lti_integration is required"
   end
+  test "should not allow duplicate subject/lti_integration_id" do
+    lti_user_identity = create(:lti_user_identity)
+    invalid_identity = build(:lti_user_identity, lti_integration_id: lti_user_identity.lti_integration.id, subject: lti_user_identity.subject)
+    refute invalid_identity.valid?
+    assert_includes invalid_identity.errors.full_messages, "Subject has already been taken"
```

Comment:
nit: use single quotes for strings that do not contain any interpolations:
- https://github.com/rubocop/ruby-style-guide?tab=readme-ov-file#single-quote
- https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#ruby-string-quotes

```suggestion
    assert_includes invalid_identity.errors.full_messages, 'Subject has already been taken'
```